e2e/machine_migration_test.go

```diff
@@ -111,6 +113,275 @@ var _ = Describe("[sig-cluster-lifecycle][OCPFeatureGate:MachineAPIMigration] Ma
+						if err := json.Unmarshal(machine.Spec.ProviderSpec.Value.Raw, &providerSpec); err != nil {
+							Fail(fmt.Sprintf("Failed to unmarshal provider spec: %v", err))
+						}
+						providerSpec.Subnet.ID = ptr.To("subnet-different123")
+						rawProviderSpec, err := json.Marshal(providerSpec)
+						if err != nil {
+							Fail(fmt.Sprintf("Failed to marshal provider spec: %v", err))
+						}
+						machine.Spec.ProviderSpec.Value.Raw = rawProviderSpec
+					}
+				})
+			})
+
+			It("should prevent modification of SecurityGroups in non-authoritative MAPI Machine", func() {
+				verifyFieldModificationPrevented(cl, testMapiMachine.Name, "SecurityGroups", func(machine *machinev1beta1.Machine) {
+					if machine.Spec.ProviderSpec.Value != nil {
+						var providerSpec machinev1beta1.AWSMachineProviderConfig
+						if err := json.Unmarshal(machine.Spec.ProviderSpec.Value.Raw, &providerSpec); err != nil {
+							Fail(fmt.Sprintf("Failed to unmarshal provider spec: %v", err))
+						}
+						providerSpec.SecurityGroups = append(providerSpec.SecurityGroups, machinev1beta1.AWSResourceReference{
+							ID: ptr.To("sg-different123"),
+						})
+						rawProviderSpec, err := json.Marshal(providerSpec)
+						if err != nil {
+							Fail(fmt.Sprintf("Failed to marshal provider spec: %v", err))
+						}
+						machine.Spec.ProviderSpec.Value.Raw = rawProviderSpec
+					}
+				})
+			})
+
+			It("should prevent modification of Tags in non-authoritative MAPI Machine", func() {
+				verifyFieldModificationPrevented(cl, testMapiMachine.Name, "Tags", func(machine *machinev1beta1.Machine) {
+					if machine.Spec.ProviderSpec.Value != nil {
+						var providerSpec machinev1beta1.AWSMachineProviderConfig
+						if err := json.Unmarshal(machine.Spec.ProviderSpec.Value.Raw, &providerSpec); err != nil {
+							Fail(fmt.Sprintf("Failed to unmarshal provider spec: %v", err))
+						}
+						providerSpec.Tags = append(providerSpec.Tags, machinev1beta1.TagSpecification{
+							Name:  "test-tag",
+							Value: "test-value",
+						})
+						rawProviderSpec, err := json.Marshal(providerSpec)
+						if err != nil {
+							Fail(fmt.Sprintf("Failed to marshal provider spec: %v", err))
+						}
+						machine.Spec.ProviderSpec.Value.Raw = rawProviderSpec
+					}
+				})
+			})
+
+			It("should prevent modification of capacityReservationId in non-authoritative MAPI Machine", func() {
+				verifyFieldModificationPrevented(cl, testMapiMachine.Name, "capacityReservationId", func(machine *machinev1beta1.Machine) {
+					if machine.Spec.ProviderSpec.Value != nil {
+						var providerSpec machinev1beta1.AWSMachineProviderConfig
+						if err := json.Unmarshal(machine.Spec.ProviderSpec.Value.Raw, &providerSpec); err != nil {
+							Fail(fmt.Sprintf("Failed to unmarshal provider spec: %v", err))
+						}
+						providerSpec.CapacityReservationID = "cr-different123456789"
+						rawProviderSpec, err := json.Marshal(providerSpec)
```

Comment:
Is this duplicated work? Don't we do this in `updateMachineProviderSpec`? Can we use that or a similar helper func here?